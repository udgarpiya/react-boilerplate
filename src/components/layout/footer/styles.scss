.footer-component {
  &__main-container {
    padding: 0 15px;
    height: 50px;
    justify-content: center;
    display: flex;
    align-items: center;
    z-index: 1;
  }
}