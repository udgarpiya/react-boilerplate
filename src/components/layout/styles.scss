.layout-component {
  &__children-container {
    min-height: calc(100vh - 100px);
  }
}