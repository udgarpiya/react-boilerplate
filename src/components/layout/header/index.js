import React from 'react'
import './styles.scss'

const Header = () => {
  return (
    <div className="header-component__main-container">
      <h1>Header Component</h1>
    </div>
  )
}

export { Header }
export default Header
