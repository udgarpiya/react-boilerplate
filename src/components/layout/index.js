import React from 'react'
import PropTypes from 'prop-types'

import Header from './header'
import Footer from './footer'
import CustomHead from '../custom-head'
import './styles.scss'

const Layout = ({ children, title, description }) => {
  return (
    <div className="layout-component__main-component">
      <CustomHead title={title} description={description} />
      <Header />
      <div className="layout-component__children-container">{children}</div>
      <Footer />
    </div>
  )
}

Layout.defaultProps = {
  title: '',
  description: '',
}

Layout.propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string,
  description: PropTypes.string,
}

export { Layout }
export default Layout
