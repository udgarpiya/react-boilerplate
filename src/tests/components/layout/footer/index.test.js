import React from 'react'
import { shallow } from 'enzyme'
import { Footer } from 'components/layout/footer'

describe('Footer Component', () => {
  it('It should have footer component and text to be "© 2020"', () => {
    const component = shallow(<Footer />)
    const FooterMain = component.find('.footer-component__main-container')

    expect(FooterMain.length).toBe(1)
    expect(FooterMain.text()).toBe('© 2020')
    expect(component).toMatchSnapshot()
  })
})
