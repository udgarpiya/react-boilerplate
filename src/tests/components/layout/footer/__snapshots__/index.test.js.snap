// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component It should have footer component and text to be "© 2020" 1`] = `
<div
  className="footer-component__main-container"
>
  © 
  2020
</div>
`;
